controllers/components/kserve/kserve_controller_actions.go

```diff
@@ -275,5 +275,20 @@ func setStatusFields(ctx context.Context, rr *odhtypes.ReconciliationRequest) er
 	}
 
 	k.Status.DefaultDeploymentMode = ddm
+
+	serviceMeshEnabled := false
+	if rr.DSCI.Spec.ServiceMesh != nil {
+		if rr.DSCI.Spec.ServiceMesh.ManagementState == operatorv1.Managed || rr.DSCI.Spec.ServiceMesh.ManagementState == operatorv1.Unmanaged {
+			serviceMeshEnabled = true
+		}
+	}
+	serverlessEnabled := false
```

Comment:
nit: this can probably be simplified by

```go
serverlessEnabled := k.Spec.Serving.ManagementState == operatorv1.Managed || k.Spec.Serving.ManagementState == operatorv1.Unmanaged
```